aptos-move/aptos-vm/src/move_vm_ext/warm_vm_cache.rs

```diff
@@ -0,0 +1,162 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use crate::{counters::TIMER, move_vm_ext::AptosMoveResolver};
+use aptos_framework::natives::code::PackageRegistry;
+use aptos_infallible::RwLock;
+use aptos_metrics_core::TimerHelper;
+use aptos_types::on_chain_config::OnChainConfig;
+use bytes::Bytes;
+use move_binary_format::errors::{Location, PartialVMError, VMResult};
+use move_core_types::{
+    account_address::AccountAddress,
+    ident_str,
+    identifier::Identifier,
+    language_storage::{ModuleId, CORE_CODE_ADDRESS},
+    vm_status::StatusCode,
+};
+use move_vm_runtime::{config::VMConfig, move_vm::MoveVM, native_functions::NativeFunction};
+use once_cell::sync::Lazy;
+use std::collections::HashMap;
+
+const WARM_VM_CACHE_SIZE: usize = 8;
+
+pub(crate) struct WarmVmCache {
+    cache: RwLock<HashMap<WarmVmId, MoveVM>>,
```

Comment:
I know the size won't be large, but why still not us eDashMap? It's supposed to be a better version of RwLock<HashMap<...>>, and same interfaces as HashMap

Reply:
I suspect another layer of indirection makes it a bit slower? We don't need the added concurrency most likely.

Reply:
makes sense